mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala

```diff
@@ -61,20 +79,58 @@ class LogisticRegressionModel (
 
   override protected def predictPoint(dataMatrix: Vector, weightMatrix: Vector,
       intercept: Double) = {
-    val margin = weightMatrix.toBreeze.dot(dataMatrix.toBreeze) + intercept
-    val score = 1.0 / (1.0 + math.exp(-margin))
-    threshold match {
-      case Some(t) => if (score > t) 1.0 else 0.0
-      case None => score
+    // If dataMatrix and weightMatrix have the same dimension, it's binary logistic regression.
+    if (dataMatrix.size == weightMatrix.size) {
+      val margin = dot(weights, dataMatrix) + intercept
+      val score = 1.0 / (1.0 + math.exp(-margin))
+      threshold match {
+        case Some(t) => if (score > t) 1.0 else 0.0
+        case None => score
+      }
+    } else {
+      val dataWithBiasSize = weightMatrix.size / (nClasses - 1)
+      val dataWithBias = if (dataWithBiasSize == dataMatrix.size) {
+        dataMatrix
+      } else {
+        assert(dataMatrix.size + 1 == dataWithBiasSize)
+        MLUtils.appendBias(dataMatrix)
```

Comment:
Create a matrix `W` and the intercept vector in constructor and use gemv call here to compute margin `Wx + b`.